plugin/pkg/managedseed/validator/admission.go

```diff
@@ -0,0 +1,461 @@
+		spec.Backup.Provider = shoot.Spec.Provider.Type
+	}
+
+	// Initialize and validate DNS and ingress
+	ingressDomain := fmt.Sprintf("%s.%s", common.IngressPrefix, *(shoot.Spec.DNS.Domain))
+	if spec.Ingress != nil {
+		if gardencorehelper.NginxIngressEnabled(shoot.Spec.Addons) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("ingress"), fmt.Sprintf("seed ingress controller cannot be enabled on shoot %s", kutil.ObjectName(shoot))))
+		}
+
+		if spec.DNS.Provider == nil {
+			dnsProvider, err := v.getSeedDNSProvider(shoot)
+			if err != nil {
+				if apierrors.IsInternalError(err) {
+					return allErrs, err
+				}
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("ingress"), spec.Ingress, err.Error()))
+			}
+			spec.DNS.Provider = dnsProvider
+		}
+
+		if spec.Ingress.Domain == "" {
+			spec.Ingress.Domain = ingressDomain
+		} else if spec.Ingress.Domain != ingressDomain {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ingress", "domain"), spec.Ingress.Domain, fmt.Sprintf("seed ingress domain must be equal to shoot DNS domain %s", ingressDomain)))
+		}
+	} else {
+		if (spec.DNS.IngressDomain == nil || *spec.DNS.IngressDomain == "") && gardencorehelper.NginxIngressEnabled(shoot.Spec.Addons) {
+			spec.DNS.IngressDomain = &ingressDomain
+		} else if !strings.HasSuffix(*spec.DNS.IngressDomain, *shoot.Spec.DNS.Domain) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("dns", "ingressDomain"), spec.DNS.IngressDomain, fmt.Sprintf("seed ingress domain must be a subdomain of shoot DNS domain %s", *shoot.Spec.DNS.Domain)))
+		}
+	}
+
+	// Initialize and validate networks
+	if spec.Networks.Nodes == nil {
+		spec.Networks.Nodes = shoot.Spec.Networking.Nodes
+	} else if shoot.Spec.Networking.Nodes != nil && *spec.Networks.Nodes != *shoot.Spec.Networking.Nodes {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "nodes"), spec.Networks.Nodes, fmt.Sprintf("seed nodes CIDR must be equal to shoot nodes CIDR %s", *shoot.Spec.Networking.Nodes)))
+	}
+	if spec.Networks.Pods == "" && shoot.Spec.Networking.Pods != nil {
+		spec.Networks.Pods = *shoot.Spec.Networking.Pods
+	} else if shoot.Spec.Networking.Pods != nil && spec.Networks.Pods != *shoot.Spec.Networking.Pods {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "pods"), spec.Networks.Pods, fmt.Sprintf("seed pods CIDR must be equal to shoot pods CIDR %s", *shoot.Spec.Networking.Pods)))
+	}
+	if spec.Networks.Services == "" && shoot.Spec.Networking.Services != nil {
+		spec.Networks.Services = *shoot.Spec.Networking.Services
+	} else if shoot.Spec.Networking.Services != nil && spec.Networks.Services != *shoot.Spec.Networking.Services {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "services"), spec.Networks.Pods, fmt.Sprintf("seed services CIDR must be equal to shoot services CIDR %s", *shoot.Spec.Networking.Services)))
+	}
+
+	// Initialize and validate provider
+	if spec.Provider.Type == "" {
```

Comment:
Unfortunately, if one creates/applies such a `ManagedSeed` manifest via kubectl
```yaml
spec:
  seedTemplate:
    spec:
      provider: {}
```
one has to use `--validate=false`:

```
$ k create -f local/managedseed/55-managedseed-seedtemplate.yaml
error: error validating "local/managedseed/55-managedseed-seedtemplate.yaml": error validating data: [ValidationError(ManagedSeed.spec.seedTemplate.spec.provider): missing required field "type" in com.github.gardener.gardener.pkg.apis.core.v1beta1.SeedProvider, ValidationError(ManagedSeed.spec.seedTemplate.spec.provider): missing required field "region" in com.github.gardener.gardener.pkg.apis.core.v1beta1.SeedProvider]; if you choose to ignore these errors, turn validation off with --validate=false
```

because the `provider.{type,region}` are not marked as optional.

I think, it would be nice to make them optional in the OpenAPI spec (by adding the `+optional` tag) to allow such `ManagedSeed` manifests and leverage all the defaulting by this admission plugin.
But, it doesn't need to be done in this PR. Just noting it down.

Reply:
You could still specify:

```yaml
spec:
  seedTemplate:
    spec:
      provider:
        type: ""
        region: ""
```

Are you sure you want to change the `Seed` spec just to avoid writing the above? There was a reason provider type and region were not initially optional, and I am not sure we want to change this at this point.

And BTW, even if we do it for `provider`, we still have the same issue with `backup`, `dns`, and `networks`. The full "empty" `seedTemplate` looks like this:

```
spec:
  seedTemplate:
    spec:
      backup:
        provider: ""
        secretRef: {}
      dns:
        ingressDomain: ""
      networks:
        pods: ""
        services: ""
      provider:
        type: ""
        region: ""
```

All of the above will be correctly defaulted, but they have to be specified as empty values because they are mandatory in the API. Making `provider: {}` possible is on this background only a minor improvement. Making `seedTemplate: {}` possible would be the real improvement, but it would require making all fields above non-mandatory. Honestly, I wouldn't do this just for this reason. Even if there is a strong vote to change the `Seed` API in this direction, let's please do this in a separate PR, not in this one.

Reply:
> Even if there is a strong vote to change the `Seed` API in this direction, let's please do this in a separate PR, not in this one.

Yes, sure. Also written this in my comment above.

> Are you sure you want to change the `Seed` spec just to avoid writing the above? There was a reason provider type and region were not initially optional, and I am not sure we want to change this at this point.

Well, the implications of adding an `+optional` tag just makes the field optional in the sense of OpenAPI spec. But it can still be forbidden by validation when not set under circumstances (e.g. in a Seed, but in a SeedTemplate OTOH and might be optional).
Anyways, let's continue with this PR. This is only a minor remark in UX, but can still be improved later on.